cmd/cli/mesh_list.go

```diff
@@ -63,113 +62,18 @@ func newMeshList(out io.Writer) *cobra.Command {
-				}
-			}
-		}
-
-		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\n", m, ns, strings.Join(x["Pods"], ","), v, strings.Join(smiList, ","))
-	}
+	fmt.Fprint(w, getPrettyPrintedMeshInfoList(meshInfoList, false))
 	_ = w.Flush()
 
 	return nil
 }
-
-// getNamespacePods returns a map of controller pods
-func getNamespacePods(clientSet kubernetes.Interface, m string, ns string) map[string][]string {
-	x := make(map[string][]string)
-
-	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"app": constants.OSMControllerName}}
-	listOptions := metav1.ListOptions{
-		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
-	}
-	pods, _ := clientSet.CoreV1().Pods(ns).List(context.TODO(), listOptions)
-
-	for pno := 0; pno < len(pods.Items); pno++ {
-		x["Pods"] = append(x["Pods"], pods.Items[pno].GetName())
-	}
-
-	return x
-}
-
-// getControllerDeployments returns a list of Deployments corresponding to osm-controller
-func getControllerDeployments(clientSet kubernetes.Interface) (*v1.DeploymentList, error) {
-	deploymentsClient := clientSet.AppsV1().Deployments("") // Get deployments from all namespaces
-	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"app": constants.OSMControllerName}}
-	listOptions := metav1.ListOptions{
-		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
-	}
-	return deploymentsClient.List(context.TODO(), listOptions)
-}
-
-// getMeshNames returns a set of mesh names corresponding to meshes within the cluster
-func getMeshNames(clientSet kubernetes.Interface) mapset.Set {
-	meshList := mapset.NewSet()
-
-	deploymentList, _ := getControllerDeployments(clientSet)
-	for _, elem := range deploymentList.Items {
-		meshList.Add(elem.ObjectMeta.Labels["meshName"])
-	}
-
-	return meshList
-}
-
-func getSupportedSmiForControllerPod(pod string, namespace string, cmd *meshListCmd) (map[string]string, error) {
-	dialer, err := k8s.DialerToPod(cmd.config, cmd.clientSet, pod, namespace)
-	if err != nil {
-		return nil, err
-	}
-
-	portForwarder, err := k8s.NewPortForwarder(dialer, fmt.Sprintf("%d:%d", cmd.localPort, constants.OSMHTTPServerPort))
-	if err != nil {
-		return nil, errors.Errorf("Error setting up port forwarding: %s", err)
-	}
-
-	var smiSupported map[string]string
-
-	err = portForwarder.Start(func(pf *k8s.PortForwarder) error {
-		defer pf.Stop()
-		url := fmt.Sprintf("http://localhost:%d%s", cmd.localPort, constants.HTTPServerSmiVersionPath)
-
-		// #nosec G107: Potential HTTP request made with variable url
-		resp, err := http.Get(url)
-		if err != nil {
-			return errors.Errorf("Error fetching url %s: %s", url, err)
-		}
-
-		if err := json.NewDecoder(resp.Body).Decode(&smiSupported); err != nil {
-			return errors.Errorf("Error rendering HTTP response: %s", err)
-		}
-		return nil
-	})
-	if err != nil {
-		return nil, errors.Errorf("Error retrieving supported SMI versions for pod %s in namespace %s: %s", pod, namespace, err)
-	}
-
-	return smiSupported, nil
-}
```

Comment:
FYI: Moved these from `cmd/cli/mesh_list.go` to `cmd/cli/util.go` as multiple cli commands use these helper functions.